students/Net_Michael/lesson4/mailroom2.py

```diff
@@ -0,0 +1,132 @@
+
+        print("Name\t\t\tTotal Donation\t\tNum Gifts\tAverage Gift")
+        for jj in range(ll):
+            donation = [ float(item) for index, item in enumerate(",".join(all_donations[jj]).split(","))]
+            donor_name = eval(donor_names[jj])
+            total_donation = sum(donation)
+            num_gifts = len(donation)
+            avg_donation = total_donation/num_gifts
+            print ('{:23}'.format(donor_name),
+                    '${:^6,.2f}'.format(total_donation),
+                    '{:20}'.format(num_gifts),
+                    '{:14}'.format(""), 
+                    '${:^6,.2f}'.format(avg_donation))
+    elif(thank_all.lower() == "yes"):   
+        for jj in range(ll):
+            donation = [ float(item) for index, item in enumerate(",".join(all_donations[jj]).split(","))]
+            donor_name = eval(donor_names[jj])
+            total_donation = sum(donation)
+            print('''
+            
+                  Dear {}, 
+                  
+                  Thank you for your very kind donation of ${:,.2f}.
+                  
+                  It will be put to very good use. 
+       
+                                        Sincerely,
+                                          -The Team'''.format(donor_name,total_donation)) 
+    elif(thank_one.lower() == "yes"):
+        quest = input("Do you have the donor name? (yes/no)" )
+        quest.lower()
+        if quest == "no":
+            print("Please select full name from report")
+        if quest == "yes":
+            donor_sel = input("Please give first and last name").title()
+            print(" If you don't see the letter for {}, please check the report".format(donor_sel))
+            for jj in range(ll):
+                donation = [ float(item) for index, item in enumerate(",".join(all_donations[jj]).split(","))]
+                donor_name = eval(donor_names[jj])
+                if donor_name in donor_sel:
+                    dd = len(donation)-1
+                    total_donation = donation[dd]
+                    print('''
+                    
+                    Dear {}, 
+                    Thank you for your very kind donation of ${:,.2f}.                  
+                    
+                    It will be put to very good use. 
+       
+                                        Sincerely,
+                                          -The Team'''.format(donor_name,total_donation))
+          
+    os.chdir(report_wd)
+  
+# In[6]:
+
+
+# var = 1 is for donors total summary donation Report
+# var = 2 is for List of individual donations
+# var = 3 is for send a Thank You to a single donor
+# var = 4 is for send Thank you letters to all donors
+        
+donors_dir_path = '/Users/Netsanet/Desktop/UW_courses/UWpython/Self_Paced-Online/students/Net_Michael/session04/donors'
+if __name__ == "__main__":
+    prompt(var = 1)
+    prompt(var = 2)
+    prompt(var = 3)
+    prompt(var = 4)
+    prompt(var = 5)
```

Comment:
@NetMic - I'm having a hard time running this code.  It needs some refactoring to make it executable.  I'll give partial credit, but please revise and submit for additional review and full credit.  This assignment is the basis for several iterations, so it's important to clean this up a bit.  Reach out if questions.

Reply:
Hi charles,

Is there a way, you can be online for a chat for live questions?